### ISSUE TITLE
fix pyarrow CVEs in kedro image

### ISSUE THREAD

**Issue by Nrbassili** [cfpb/hmda-data-pipelines#5]
https://github.cfpb.gov/HMDA-Operations/kedro-etl-pipeline/issues/55

**Comment by Nrbassili** [comment]
pip install --upgrade pyarrow>=14.0.1 
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pandas-gbq 0.17.9 requires pyarrow<10.0dev,>=3.0.0, but you have pyarrow 15.0.2 which is incompatible.
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

**Comment by Nrbassili** [comment]
Updated
`kedro==1.18.12` from requirements.txt, to `kedro` (defaults to version 0.19.3)
`SQLAlchemy<2` from requirements.txt, to `SQLAlchemy` (defaults to version 2.0.28)

**Comment by rkovalik-raft** [comment]
Resolved with Kedro version upgrade to 0.19.9

```
docker scout cves 626560329871.dkr.ecr.us-east-1.amazonaws.com/hmda/kedro-etl-pipeline:kedro-latest --only-severity=critical,high
INFO New version 1.15.1 available (installed version is 0.16.1)
    ✓ SBOM of image already cached, 774 packages indexed
    ✗ Detected 1 vulnerable package with 2 vulnerabilities

   0C     2H     0M     0L  setuptools 58.1.0
pkg:pypi/setuptools@58.1.0

    ✗ HIGH CVE-2022-40897 [Inefficient Regular Expression Complexity]
      https://scout.docker.com/v/CVE-2022-40897?s=github&n=setuptools&t=pypi&vr=%3C65.5.1
      Affected range : <65.5.1                                                          
      Fixed version  : 65.5.1                                                           
      CVSS Score     : 8.7                                                              
      CVSS Vector    : CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:H/SC:L/SI:L/SA:N  
    
    ✗ HIGH CVE-2024-6345 [Improper Control of Generation of Code ('Code Injection')]
      https://scout.docker.com/v/CVE-2024-6345?s=github&n=setuptools&t=pypi&vr=%3C70.0.0
      Affected range : <70.0.0                                                          
      Fixed version  : 70.0.0                                                           
      CVSS Score     : 7.5                                                              
      CVSS Vector    : CVSS:4.0/AV:N/AC:L/AT:P/PR:N/UI:A/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N  
    


2 vulnerabilities found in 1 package
  LOW       0  
  MEDIUM    0  
  HIGH      2  
  CRITICAL  0 
```
